packages/bundler-plugin-core/src/debug-id-upload.ts

```diff
@@ -298,10 +298,28 @@ async function determineSourceMapPathFromBundle(
   const sourceMappingUrlMatch = bundleSource.match(/^\s*\/\/# sourceMappingURL=(.*)$/m);
   if (sourceMappingUrlMatch) {
     const sourceMappingUrl = path.normalize(sourceMappingUrlMatch[1] as string);
-    if (path.isAbsolute(sourceMappingUrl)) {
-      return sourceMappingUrl;
+
+    let isSupportedUrl;
+    try {
+      const url = new URL(sourceMappingUrl);
+      isSupportedUrl = url.protocol === "file:";
+    } catch {
+      isSupportedUrl = false;
+    }
+
+    let absoluteSourceMapPath;
+    if (isSupportedUrl || path.isAbsolute(sourceMappingUrl)) {
+      absoluteSourceMapPath = sourceMappingUrl;
     } else {
-      return path.join(path.dirname(bundlePath), sourceMappingUrl);
+      absoluteSourceMapPath = path.join(path.dirname(bundlePath), sourceMappingUrl);
+    }
+
+    try {
+      // Check if the file actually exists
+      await util.promisify(fs.access)(absoluteSourceMapPath);
```

Comment:
l: any reason we use this over `fs.promises.access`?

(don't feel obligated to change it, I'm primarily just curious)

Reply:
No not really. I think originally I used it because it was available in earlier node versions I believe.